pkg/profile/artifact_test.go

```diff
@@ -678,4 +681,237 @@ var _ = Describe("Profile", func() {
+			Expect(hasCorrectFilePerms(profileFile)).To(BeTrue())
+			Expect(hasCorrectFilePerms(artifactFile)).To(BeTrue())
+			Expect(hasCorrectFilePerms(artifactFileDeployment)).To(BeTrue())
+
+			content, err := ioutil.ReadFile(profileFile)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(content)).To(Equal(`apiVersion: weave.works/v1alpha1
+kind: ProfileInstallation
+metadata:
+  creationTimestamp: null
+  name: mySub
+  namespace: default
+spec:
+  source:
+    branch: main
+    path: weaveworks-nginx
+    url: https://github.com/weaveworks/profiles-examples
+status: {}
+`))
+
+			content, err = ioutil.ReadFile(artifactFile)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(content)).To(Equal(fmt.Sprintf(`apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
+kind: Kustomization
+metadata:
+  creationTimestamp: null
+  name: mySub-weaveworks-nginx-nginx-deployment
+  namespace: default
+spec:
+  interval: 5m0s
+  path: %s/artifacts/nginx-deployment/nginx/deployment
+  prune: true
+  sourceRef:
+    kind: GitRepository
+    name: git-repo-name
+    namespace: git-repo-namespace
+  targetNamespace: default
+status: {}
+`, tempDir)))
+		})
+	})
+
+	When("there is a flat profile repository", func() {
+		It("creates files for all artifacts", func() {
+			pSub = profilesv1.ProfileInstallation{
+				TypeMeta: profileTypeMeta,
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      subscriptionName,
+					Namespace: namespace,
+				},
+				Spec: profilesv1.ProfileInstallationSpec{
+					Source: &profilesv1.Source{
+						URL:    "https://github.com/weaveworks/nginx-profile",
+						Branch: "main",
+					},
+				},
+			}
+			p.SetProfileGetter(func(repoURL, branch, path string, gitClient git.Git) (profilesv1.ProfileDefinition, error) {
+				return profilesv1.ProfileDefinition{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "nginx",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Profile",
+						APIVersion: "packages.weave.works.io/profilesv1",
+					},
+					Spec: profilesv1.ProfileDefinitionSpec{
+						ProfileDescription: profilesv1.ProfileDescription{
+							Name:        "nginx",
+							Description: "foo",
+						},
+						Artifacts: []profilesv1.Artifact{
+							{
+								Name: "bitnami-nginx",
+								Chart: &profilesv1.Chart{
+									URL:     "https://charts.bitnami.com/bitnami",
+									Name:    "nginx",
+									Version: "8.9.1",
+								},
+							},
+						},
+					},
+				}, nil
+			})
+			tempDir, err := ioutil.TempDir("", "catalog-install")
+			Expect(err).NotTo(HaveOccurred())
+			maker := profile.NewProfilesArtifactsMaker(profile.MakerConfig{
+				ProfileName:      "generate-test",
+				GitClient:        fakeGitClient,
+				RootDir:          tempDir,
+				GitRepoNamespace: gitRepoNamespace,
+				GitRepoName:      gitRepoName,
+			})
+			artifacts, err := maker.MakeArtifacts(pSub)
+			Expect(err).NotTo(HaveOccurred())
+			err = maker.GenerateArtifactsOutput(artifacts, pSub)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
if we do keep these funcs separate its better to not test one functions in another functions suite, so instead we could fake out the `Artifacts` struct. Might also make the test smaller/cleaner :smile:

Reply:
Yeah, agreed